pkg/k8s/apis/cilium.io/v2alpha1/flowlog.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:onlyVerbs=get,list,watch
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium},singular="ciliumflowlog",path="ciliumflowlogs",scope="Cluster",shortName={cfl,ciliumfl}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",description="Time duration since creation of CiliumFlow Logging",name="Age",type=date
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// CiliumFlowLog configures a flow log task for Cilium agents.
+//
+// +deepequal-gen=false
+type CiliumFlowLog struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the desired specification/configuration of the flow log.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Spec is immutable"
+	Spec FlowLogSpec `json:"spec"`
+
+	// Status defines the realized specification/configuration and status
+	// of the flow log.
+	//
+	// +kubebuilder:validation:Optional
+	Status FlowLogStatus `json:"status,omitempty"`
+}
+
+// FlowLogSpec defines configuration of a flow log task.
+//
+// +deepequal-gen=false
+type FlowLogSpec struct {
+	// List of field names that will be kept in the log output.
+	//
+	// +kubebuilder:validation:Optional
+	FieldMask []string `json:"fieldmask,omitempty"`
+
+	// Filters specifies flows to include in the log output. Flow passes the
+	// filter if any of the entries match. Only Flows matching the filter, but
+	// not matching exclude-filter will be logged. The check is disabled when
+	// empty.
+	//
+	// +kubebuilder:validation:Optional
+	Filters []*flowpb.FlowFilter `json:"filters.omitempty"`
+
+	// ExcludeFilters specifies flows to exclude from the log output. Flow
+	// passes the filter if any of the entries match. Only Flows matching the
+	// filter, but not matching exclude-filter will be logged. The check is
+	// disabled when empty.
+	//
+	// +kubebuilder:validation:Optional
+	ExcludeFilters []*flowpb.FlowFilter `json:"exludefilters,omitempty"`
+
+	// Expires specifies the time when log will stop. Empty means
+	// that flow log won't stop until this object is deleted.
+	//
+	// +kubebuilder:validation:Format=date-time
+	// +kubebuilder:validation:Optional
+	Expires *metav1.Time `json:"expires,omitempty"`
+}
+
+// FlowLogStatus is a status of a flow log task.
+type FlowLogStatus struct{}
```

Comment:
Without status, we can't tell if it is working without digging into the node files.

We should consider some status, like number of flows logged per node, and conditions for whether the node is failing to log (maybe due to permissions or storage issues).

Reply:
Agree. I would expect at least the `Conditions` to indicate if the flow filter has been programmed successfully.

Reply:
Who would own the object in order to update it?

Reply:
I don't know what status would be present without seeing the implementation.

I imagine there would be status associated with each node so there could be a Nodes slice where each node owns and updates its own item in the slice.

We would have to have some central controller which could clean up stale entries for nodes which don't exist anymore or aren't running anetd.

Reply:
I wanted to avoid having a central controller for this. I thought that the state would be exposed by metrics:
```
flow_log_active_task{name="just-an-example"} 1
flow_log_processed_flows{name="just-an-example"} 1
```

Does it need to be figured out in this PR or can be added later when we will have more details from the implementation?